serving/docs/lmi_new/deployment_guide/deploying-your-endpoint.md

```diff
@@ -1,6 +1,140 @@
+
+If your model artifacts are also stored in S3, then you can either upload the `serving.properties` file under the same object prefix, or in a separate location.
+
+If you upload the `serving.properties` file under the same object prefix as your model artifacts, you should not specify `option.model_id` in the configuration.
+
+For example, your directory containing model artifacts and serving.properties may look like this (flat structure):
+```shell
+my-lmi-model/
+  - serving.properties
+  - config.json
+  - model-0001-of-0002.safetensors
+  - model-0002-of-0002.safetensors
+  - model.safetensors.index.json
+  - tokenizer.json
+  - tokenizer.model
+```
+
+If you upload the `serving.properties` to a different location than your model artifacts, make sure `option.model_id` points to the S3 URI of the object prefix (e.g. `option.model_id=s3://my-bucket/my-model-artifacts/`).
+
+Here is the sample code you can use to upload your configuration to s3:
+```shell
+# Assumes that the serving.properties file is stored in a local folder called my-lmi-model/
+# The my-lmi-model folder my also contain your model artifacts
+tar czvf my-lmi-model.tar.gz -C my-lmi-model .
+aws s3 cp my-lmi-model.tar.gz s3://my-bucket/lmi-models/ 
+```
+
+Here is the code to deploy your model on SageMaker using LMI:
+```python
+import sagemaker
+
+# Your IAM role that provides access to SageMaker and S3. 
+# See https://docs.aws.amazon.com/sagemaker/latest/dg/automatic-model-tuning-ex-role.html if running on a SageMaker notebook
+iam_role = "my-sagemaker-role"
+# manages interactions with the sagemaker apis
+sagemaker_session = sagemaker.session.Session()
+# region is needed to retrieve the lmi container
+region = sagemaker_session._region_name
+# get the lmi image uri
+container_uri = sagemaker.image_uris.retrieve(framework="djl-deepspeed", version="0.26.0", region=region)
+# create a unique endpoint name
+endpoint_name = sagemaker.utils.name_from_base("my-lmi-endpoint")
+# s3 location of the tar.gz containing your configuration and potentially model artifacts
+model_tar = "s3://my-bucket/lmi-models/my-lmi-model.tar.gz"
+# instance type you will deploy your model to
+instance_type = "ml.g5.12xlarge"
+
+# create your SageMaker Model
+model = sagemaker.Model(image_uri=container_uri, model_data=model_tar, role=iam_role)
+# deploy your model
+model.deploy(
+    instance_type=instance_type,
+    initial_instance_count=1,
+    endpoint_name=endpoint_name,
+)
+
+# Get a predictor for your endpoint
+predictor = sagemaker.Predictor(
+    endpoint_name=endpoint_name,
+    sagemaker_session=sagemaker_session,
+    serializer=sagemaker.serializers.JSONSerializer(),
+    deserializer=sagemaker.deserializers.JSONDeserializer(),
+)
+
+# Make a prediction with your endpoint
+outputs = predictor.predict({
+    "inputs": "The meaning of life is", 
+    "parameters": {"do_sample": True, "max_new_tokens": 256}
+})
+```
+
+## Configuration - environment variables
```

Comment:
Do we recommend one approach vs other between serving.properties and env var? I personally feel env vars is simple and we should put that at the top. (and we should get rid of `SERVING_LOAD_MODELS` env var requirement)